### PR TITLE
API: Add screenshots link to units

### LINKED ISSUE #6080: Screenshot link for units (API)
Hello everybody,

I use the Weblate API for the external recording app. In the project, sometimes Voice Over artist wants to see the screenshot before the recording. I am already using the "Screenshot feature" in the Weblate. But we can not take the screenshots links special for the units via API. So, it can be nice to see a `screenshot_link` field for these endpoints; 

```/api/translations/PROJECT/COMPONENT/LANGUAGE/units/```
```/api/units/UNIT_ID```

Thanks!

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #6080

### CODE REVIEW

**Comment by prateekj117** [comment]
Hey @nijel , let me know if the implementation looks good and I will go ahead with the tests and docs changes then.

File: weblate/api/views.py [hunk 1]
```diff
@@ -1212,6 +1212,45 @@ def get_serializer(self, instance, *args, **kwargs):
...
+            raise ValidationError({"screenshot_id": "This field is required."})
+
+        try:
+            screenshot = obj.translation.screenshot_set.get(
+                pk=int(request.data["screenshot_id"])
+            )
+        except (Screenshot.DoesNotExist, ValueError) as error:
+            raise ValidationError({"screenshot_id": str(error)})
+
+        obj.units.add(screenshot)
```
Comment (nijel) [inline weblate/api/views.py:1232]: This looks wrong - `obj` is `Unit` here, it probably shoule be:

```suggestion
        screenshot.units.add(obj)
```

File: weblate/api/views.py [hunk 2]
```diff
@@ -1212,6 +1212,45 @@ def get_serializer(self, instance, *args, **kwargs):
     def get_queryset(self):
         return Unit.objects.filter_access(self.request.user).order_by("id")
 
+    @action(detail=True, methods=["post"])
+    def screenshots(self, request, **kwargs):
+        obj = self.get_object()
+
+        if not request.user.has_perm("unit.edit", obj.translation):
```
Comment (nijel) [inline weblate/api/views.py:1219]: It should check the screenshot permission:

```suggestion
        if not request.user.has_perm("screenshot.edit", obj.translation):
```

File: weblate/api/views.py [hunk 3]
```diff
@@ -1212,6 +1212,45 @@ def get_serializer(self, instance, *args, **kwargs):
...
+        return Response(serializer.data, status=HTTP_200_OK)
+
+    @action(
+        detail=True,
+        methods=["delete"],
+        url_path="screenshots/(?P<screenshot_id>[0-9]+)",
+    )
+    def delete_screenshots(self, request, pk, screenshot_id):
+        obj = self.get_object()
+        if not request.user.has_perm("unit.edit", obj.translation):
```
Comment (nijel) [inline weblate/api/views.py:1244]:
```suggestion
        if not request.user.has_perm("screenshot.edit", obj.translation):
```

**Pull request closed without merging**